src/middlewares/validation.ts

```diff
@@ -276,4 +277,66 @@ const isGoogleEnabled = async (req: any, res: Response, next: NextFunction) => {
     if (req.user.isGoogleAccount) return res.status(httpStatus.UNAUTHORIZED).json({ status: httpStatus.UNAUTHORIZED, message: "This is google account, please login with google" })
     return next();
 }
-export { validation, isUserExist, isAccountVerified, verifyUserCredentials, isUsersExist, isProductExist, isShopExist, transformFilesToBody, credential, isSessionExist, verifyUser,isGoogleEnabled,isUserEnabled,isUserVerified };
\ No newline at end of file
+const productsByCategory = async (req: Request, res: Response, next: NextFunction) => {
+    const category = req.query.category
+    if (category) {
+        const products = await productRepositories.getAvailableProductsByAttributes("category", category)
+        return res.status(httpStatus.OK).json({ status: httpStatus.OK, data: products })
+    }
+    return next()
+}
+const sellersShop = async (req: any, res: Response, next: NextFunction) => {
+    try {
+        const user = req.user
+        const shop = await productRepositories.findByModelsAndAttributes(Shops, "userId", "userId", user.id, user.id);
+        if (!shop) {
+            return res.status(httpStatus.NOT_FOUND).json({ error: "Shop doesn't exists" });
+        }
+        req.shop = shop;
+        return next()
+    } catch (error) {
+        return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({ status: httpStatus.INTERNAL_SERVER_ERROR, error: error.message });
+
+    }
+}
+
+
+const isSellerShopExist = async (req: any, res: Response, next: NextFunction) => {
+    try {
+        const shop = await productRepositories.findShopByAttributes(Shops, "userId", req.user.id)
+        if (!shop) {
+            return res.status(httpStatus.NOT_FOUND).json({ status: httpStatus.NOT_FOUND, message: "Shop not found" });
+        }
+        req.shop = shop;
+        return next();
+    } catch (error) {
+        return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({ status: httpStatus.INTERNAL_SERVER_ERROR, message: error.message });
+    }
+}
+const isProductsByCategorySelected = async (req: Request, res: Response, next: NextFunction) => {

```

Comment:
Also, 'isProductsbyCategorySelected' is essentially the same as 'productsByCategory'. It seems like you are duplicating functions.